src/jax_quantity/_register_primitives.py

```diff
@@ -891,13 +891,13 @@ def _mul_p_qq(x: Quantity, y: Quantity) -> Quantity:
 
 
 @register(lax.mul_p)
-def _mul_p_vq(x: Value, y: Quantity) -> Quantity:
-    return Quantity(lax.mul(x, y.value), unit=y.unit)
+def _mul_p_vq(x: DenseArrayValue, y: Quantity) -> Quantity:
+    return Quantity(lax.mul(x.array, y.value), unit=y.unit)
```

Comment:
Check that the `.array` is necessary. I was getting an infinite loop, but didn't investigate whether this was the best solution.